File: frontend/src/types/dto/entity/LobbyDto.ts
import type { Difficulty } from "../../enum/Difficulty"
import type { TimeLimitPreset } from "../../enum/TimeLimitPreset"
import type { LobbyPlayerDto } from "./LobbyPlayerDto"
import type { UserDto } from "./UserDto"

export type LobbyDto = {
    id: number,
    lobbyName: string,
    createdAt: string,
    difficulty: Difficulty,
    timeLimit: TimeLimitPreset,
    isActive: boolean,
    isPublic: boolean,
    inGame: boolean,
    currentGameId: number | null,
    countdownActive: boolean,
    countdownEndsAt: string,
    countdownInitiatedBy: number,
    settingsLocked: boolean,
    lobbyPlayers: LobbyPlayerDto[],
    host: UserDto
}

File: backend/src/main/java/com/github/ryand6/sudokuweb/dto/entity/LobbyDto.java
package com.github.ryand6.sudokuweb.dto.entity;

import com.github.ryand6.sudokuweb.enums.Difficulty;
import com.github.ryand6.sudokuweb.enums.TimeLimitPreset;
import lombok.*;

import java.time.Instant;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
/*
LobbyDto used for transferring necessary Lobby entity fields to frontend
*/
public class LobbyDto {

    private Long id;

    private String lobbyName;

    private Instant createdAt;

    private Difficulty difficulty;

    private TimeLimitPreset timeLimit;

    private boolean isActive;

    private boolean isPublic;

    private boolean inGame;

    private Long currentGameId;

    private boolean countdownActive;

    private Instant countdownEndsAt;

    private Long countdownInitiatedBy;

    private boolean settingsLocked;

    private Set<LobbyPlayerDto> lobbyPlayers;

    private UserDto host;

}
